client/packages/common/src/hooks/useConfirmOnLeaving/useConfirmOnLeaving.ts

```diff
@@ -1,32 +1,42 @@
-import { useCallback } from 'react';
+import { useCallback, useContext, useEffect } from 'react';
 import { BlockerFunction, useBeforeUnload, useBlocker } from 'react-router-dom';
 import { useTranslation } from '@common/intl';
+import { ConfirmationModalContext } from '@openmsupply-client/common'; // '@common/components';
 
 /** useConfirmOnLeaving is a hook that will prompt the user if they try to navigate away from,
  *  or refresh the page, when there are unsaved changes.
  * */
 export const useConfirmOnLeaving = (isUnsaved?: boolean) => {
   const t = useTranslation();
-  const confirmMessage = `${t('heading.are-you-sure')}\n${t('messages.confirm-cancel-generic')}`;
-
   const customConfirm = (onOk: () => void) => {
-    if (confirm(confirmMessage)) {
-      onOk();
-    }
+    setOnConfirm(onOk);
+    showConfirmation();
   };
 
-  useBlocker(
-    useCallback<BlockerFunction>(
-      ({ currentLocation, nextLocation }) => {
-        if (!!isUnsaved && currentLocation.pathname !== nextLocation.pathname) {
-          return !confirm(confirmMessage);
-        }
-        return false;
-      },
-      [isUnsaved]
-    )
+  const { setOpen, setMessage, setOnConfirm, setTitle } = useContext(
+    ConfirmationModalContext
   );
 
+  const showConfirmation = useCallback(() => {
+    setMessage(t('heading.are-you-sure'));
```

Comment:
manually doing this rather than using the `useConfirmationModal` hook as I had issues with that call, and having a captured enclosure around the blocker.